src/WebJobs.Extensions.SendGrid/Bindings/SendGridMessageAsyncCollector.cs

```diff
@@ -33,7 +34,7 @@ public Task AddAsync(SendGridMessage item, CancellationToken cancellationToken =
 
             SendGridHelpers.DefaultMessageProperties(item, _config, _attribute);
 
-            if (item.To == null || item.To.Length == 0)
+            if (item.Personalization?.Count == 0)
```

Comment:
I'm on board with you making whatever changes are necessary to ensure a good error experience. I agree that we don't want to duplicate their validations. I made this comment because the error message was about a missing "to" field, but your changes were such that it could happen in other cases, so the error message could be misleading.

If your experimentation indicates that we don't need to be doing these validations, then remove them.